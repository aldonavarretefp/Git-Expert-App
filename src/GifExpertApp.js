import {useState} from 'react';

import AddCategory from './components/AddCategory';
import GifGrid from './components/GifGrid';

function GifExpertApp({defaultCategories = []}) {
    const [categories, setCategories] = useState(defaultCategories);

    // const handleAdd = () => {
    //     // setCategories(['HunterXHunter', ...categories]);
    //     setCategories(cat => ['HunterXHunter',...cat]);//callback
    // }
    
    return (
        <>
            <h1>GifExpertApp</h1>
            <AddCategory setCategories={setCategories}/>
            <hr/>
            {/* <button onClick={handleAdd}>Agregar</button> */}
                <ol>
                    {
                        categories.map((category) => (
                            <GifGrid
                                key={category}
                                category={category}
                            /> //Si lo vamos a renderizar se utiliza el key
                        ))
                    }
                </ol>
        </>
    )
}

export default GifExpertApp
