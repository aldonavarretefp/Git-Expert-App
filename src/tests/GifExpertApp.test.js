import { shallow } from 'enzyme';
import React from 'react';
import GifExpertApp from '../GifExpertApp';


describe('Pruebas en GifExpertApp',()=>{
    test('debe mostrarse correctamente', () => {
        let wrapper = shallow(<GifExpertApp/>);
        expect(wrapper).toMatchSnapshot();
    });
    test('debe mostrar una lista de categorias', () => {
        let categories = ['HunterXHunter', 'OnePiece'];
        let wrapper = shallow(<GifExpertApp defaultCategories={categories}/>);
        expect(wrapper).toMatchSnapshot();
        expect(wrapper.find('GifGrid').length).toBe(categories.length);
    });
    
});