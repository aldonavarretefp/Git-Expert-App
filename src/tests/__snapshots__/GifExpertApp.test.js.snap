// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en GifExpertApp debe mostrar una lista de categorias 1`] = `
<Fragment>
  <h1>
    GifExpertApp
  </h1>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <ol>
    <GifGrid
      category="HunterXHunter"
      key="HunterXHunter"
    />
    <GifGrid
      category="OnePiece"
      key="OnePiece"
    />
  </ol>
</Fragment>
`;

exports[`Pruebas en GifExpertApp debe mostrarse correctamente 1`] = `
<Fragment>
  <h1>
    GifExpertApp
  </h1>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <ol />
</Fragment>
`;
